cmd/server/http/jwt.go

```diff
@@ -0,0 +1,134 @@
+package http
+
+import (
+	"context"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	httpinternal "github.com/lunarway/release-manager/internal/http"
+	"github.com/pkg/errors"
+)
+
+const keyNotFoundMsg = "failed to find key with key ID"
+
+type JwkCache interface {
+	Get(ctx context.Context, url string) (jwk.Set, error)
+	Refresh(ctx context.Context, url string) (jwk.Set, error)
+}
+
+type Verifier struct {
+	jwksLocation string
+	issuer       string
+	audience     string
+
+	jwkCache JwkCache
+}
+
+func NewVerifier(jwksLocation string, jwkFetchTimeout time.Duration, issuer string, audience string) (*Verifier, error) {
+	ctx := context.Background()
+
+	cache := jwk.NewCache(ctx)
+	err := cache.Register(jwksLocation, jwk.WithMinRefreshInterval(24*time.Hour))
+	if err != nil {
+		return nil, err
+	}
+	_, err = cache.Refresh(ctx, jwksLocation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Verifier{
+		jwksLocation: jwksLocation,
+		jwkCache:     cache,
+		issuer:       issuer,
+		audience:     audience,
+	}, nil
+}
+
+func ParseBearerToken(token string) (string, error) {
+	jwt := strings.TrimPrefix(token, "Bearer")
+
+	tokenParts := strings.Split(jwt, ".")
+
+	if len(tokenParts) != 3 {
+		return "", errors.New("invalid token format")
+	}
+
+	return strings.TrimSpace(jwt), nil
+}
+
+const AUTH_USER_KEY = "AUTH_USER_KEY"
+
+// authenticate authenticates the handler against a Bearer token.
+//
+// If authentication fails a 401 Unauthorized HTTP status is returned with an
+// ErrorResponse body.
+func (v *Verifier) authentication(token string) func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			bearerToken, err := ParseBearerToken(token)
+			if err != nil {
```

Comment:
This conditional is a bit hard to grok. If the server is started with a JWT as hamctl token, it will parse the request's JWT? Or? No, that is not right. The `token` variable is from the server startup arguments. 🧐

What is going on here?
Maybe it should be extracted into some named functions to make it more clear what happens in each clause?